storm/src/main/storm/mesos/NimbusMesosScheduler.java

```diff
@@ -109,6 +119,28 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
     }
   }
 
+  private void updateLogviewerState(TaskStatus status) {
+    String nodeId = status.getTaskId().getValue().split("\\" + DEFAULT_MESOS_COMPONENT_NAME_DELIMITER)[0];
+    switch (status.getState()) {
+      case TASK_STAGING: return;
+      case TASK_STARTING: return;
+      case TASK_RUNNING: return;
+      default:
+    }
+    // if it gets to this point it means logviewer terminated; update ZK with new logviewer state
```

Comment:
I see, in the case of task status 10, 12, and 13 the task may or may not still be running due to some type of network partitioning or what not. Luckily the driver has an explicit kill command we can use.